### PR TITLE
feat: Implement error recovery for the parallel runner

### PR DESCRIPTION
**Pull request by isidentical**
Depends on #423

### CODE REVIEW

File: src/fal/planner/schedule.py [hunk 1]
```diff
@@ -47,7 +47,9 @@ def create_group(
 
 @dataclass
 class Scheduler:
-    groups: list[TaskGroup]
+    groups: List[TaskGroup]
+    failed_groups: List[TaskGroup] = field(default_factory=list)
+    skipped_groups: List[TaskGroup] = field(default_factory=list)
```
Comment (isidentical) [inline src/fal/planner/schedule.py:52]: Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`. Don't know though, this version seems very simple too. Any suggestions?

Comment (chamini2) [reply src/fal/planner/schedule.py:52]: > Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`

I would prefer this, tbh. then getting `failed_groups`/`skipped_groups` can be a function that just filters. But I am also OK with whatever you prefer.

**Review by chamini2 — COMMENTED** [review]
I am kind of confused, apart from logging, what else are we achieving?

**Comment by isidentical** [comment]
> I am kind of confused, apart from logging, what else are we achieving?

The error recovery was already partially implemented, but it was prevented from functioning since scheduler raised an error whenever something failed. That part is now removed, added logging for the failed/skipped nodes and tests.

**Review by chamini2 — APPROVED** [review]

**Review by turbo1912 — APPROVED** [review]

File: src/fal/planner/executor.py [hunk 2]
```diff
@@ -13,6 +13,19 @@
...
+
+def _show_failed_groups(failed_groups: List[TaskGroup], skipped_groups: List[TaskGroup]) -> None:
+    failed_models = ", ".join(
+        model for group in failed_groups for model in group.task.model_ids
+    )
+    skipped_models = ", ".join(
+        model for group in skipped_groups for model in group.task.model_ids
+    )
+    logger.info(f"Failed calculating the following DBT models: {model_text}")
+    logger.info(f"Skipped calculating the following DBT models: {model_text}")
```
Comment (isidentical) [inline src/fal/planner/executor.py:27]:
```suggestion
    logger.info(f"Failed calculating the following DBT models: {failed_models}")
    logger.info(f"Skipped calculating the following DBT models: {skipped_models}")
```

**Review by mederka — APPROVED** [review]

**Review by turbo1912 — APPROVED** [review]

File: src/fal/planner/schedule.py (hunk 1 above)
Comment (chamini2) [reply src/fal/planner/schedule.py:52]: Are we going to change this?

**Pull request merged**